Expensify/App
# Fix: Blank screen after creating an account

<If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit.>
@marcaaron

### Details
Address the problem that no navigator is available during application initialization
We don't need to keep the entire AuthScreens navigator unmounted but just the Report screen
This way the URL still resolves to the correct report after the data is available

### Fixed Issues
Fixes #2497

### Tests


1. Start from logged out state and proceed to create a new account
2. While logged out, type and login with brand new email address so it can trigger sign up flow
3. Get validation link from e-mail. On dev I'm copying the email link and appending it to localhost - e.g. http://localhost:8080/setpassword/10024706/EHOHMPGWM
4. Type new password and click "set password"
5. After a brief loading time you should see an initial chat list in LHN containing only the Concierge chat
    - the URL in the address would be `/r` until the initial report number is resolved
    - Depending on internet speed this can be too quick to notice
    - In the video samples below it reads `/loading` but we decided to change it
6. The Concierge chat should be loaded as the active conversation

Please check the steps listed in the PR here #2346 work as well and there's no regressions

### QA Steps
Same as above

### Tested On


### Screenshots


#### Web


I'm intentionally setting a slow internet connection to test that the URL (chat) would resolve correctly even when the data comes after the App's initial render

##### Creating account over a slow internet connection
https://user-images.githubusercontent.com/12156624/116293730-db8cab80-a79f-11eb-9ace-9f8ff0d4047f.mp4

##### Login in over a slow internet connection

https://user-images.githubusercontent.com/12156624/116294035-2c9c9f80-a7a0-11eb-84d2-de53b20003f3.mp4

#### Mobile Web


##### Logging in on mobile web
https://user-images.githubusercontent.com/12156624/116302433-d7658b80-a7a9-11eb-814d-947bdeec626b.mov

#### Desktop


##### Logging in on the desktop app
https://user-images.githubusercontent.com/12156624/116306650-0251de80-a7ae-11eb-8b93-e269f128f7df.mov

#### iOS


A simulated login using the magic link after an account is created

##### Simulate a magic link to set a password for a new account

https://user-images.githubusercontent.com/12156624/116302029-61f9bb00-a7a9-11eb-9a06-58b4d59071d9.mp4

#### Android

Testing on Android to verify everything still works as before

https://user-images.githubusercontent.com/12156624/116294183-58b82080-a7a0-11eb-960c-7802634bbe6f.mp4
